cmd/anonymizer/app/writer/writer_test.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package writer
+
+import (
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/model"
+)
+
+var tags = []model.KeyValue{
+	model.Bool("error", true),
+	model.String("http.method", "POST"),
+	model.Bool("foobar", true),
+}
+
+var traceID = model.NewTraceID(1, 2)
+
+var span = &model.Span{
+	TraceID: traceID,
+	SpanID:  model.NewSpanID(1),
+	Process: &model.Process{
+		ServiceName: "serviceName",
+		Tags:        tags,
+	},
+	OperationName: "operationName",
+	Tags:          tags,
+	Logs: []model.Log{
+		{
+			Timestamp: time.Now(),
+			Fields: []model.KeyValue{
+				model.String("logKey", "logValue"),
+			},
+		},
+	},
+	Duration:  time.Second * 5,
+	StartTime: time.Unix(300, 0),
+}
+
+func TestNew(t *testing.T) {
+	nopLogger := zap.NewNop()
+	tempDir := t.TempDir()
+
+	tests := []struct {
+		name   string
+		config Config
+	}{
+		{
+			name: "successfully create writer",
+			config: Config{
+				MaxSpansCount:  10,
+				CapturedFile:   tempDir + "/captured.json",
+				AnonymizedFile: tempDir + "/anonymized.json",
+				MappingFile:    tempDir + "/mapping.json",
+			},
+		},
+		{
+			name: "captured.json doesn't exist",
+			config: Config{
+				CapturedFile:   tempDir + "/nonexistent_directory/captured.json",
+				AnonymizedFile: tempDir + "/anonymized.json",
+				MappingFile:    tempDir + "/mapping.json",
+			},
+		},
+		{
+			name: "anonymized.json doesn't exist",
+			config: Config{
+				CapturedFile:   tempDir + "/captured.json",
+				AnonymizedFile: tempDir + "/nonexistent_directory/anonymized.json",
+				MappingFile:    tempDir + "/mapping.json",
+			},
+		},
+	}
+
+	t.Run(tests[0].name, func(t *testing.T) {
+		_, err := New(tests[0].config, nopLogger)
+		require.NoError(t, err)
+	})
```

Comment:
The point of doing data driven tests is so that you don't have to repeat code like this. You can either run them in a loop, or don't bother defining array in the first place and just initialize arguments right here, in the sub-tests (I personally prefer the latter as it makes it easier to understand the stacktrace, but when there's certain repeated logic for each test then data driven makes more sense).

Reply:
Alright I will make the change.

Reply:
Made the fixes